packages/smooth_app/lib/pages/image/product_image_gallery_other_view.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:smooth_app/generic_lib/widgets/images/smooth_image.dart';
+import 'package:smooth_app/pages/image/product_image_other_page.dart';
+import 'package:smooth_app/query/product_query.dart';
+
+/// Display of the other pictures of a product.
+class ProductImageGalleryOtherView extends StatefulWidget {
+  const ProductImageGalleryOtherView({
+    required this.product,
+  });
+
+  final Product product;
+
+  @override
+  State<ProductImageGalleryOtherView> createState() =>
+      _ProductImageGalleryOtherViewState();
+}
+
+class _ProductImageGalleryOtherViewState
+    extends State<ProductImageGalleryOtherView> {
+  late final Future<List<int>> _loading = _loadOtherPics();
+
+  /// Number of columns for the grid.
+  static const int _columns = 3;
+
+  Future<List<int>> _loadOtherPics() async =>
+      OpenFoodAPIClient.getProductImageIds(
+        widget.product.barcode!,
+        uriHelper: ProductQuery.uriProductHelper,
+        user: ProductQuery.getUser(),
+      );
+
+  @override
+  Widget build(BuildContext context) {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+    final double screenWidth = MediaQuery.of(context).size.width;
+    final double squareSize = screenWidth / _columns;
+    return FutureBuilder<List<int>>(
```

Comment:
Oh no, please don't use a `FutureBuilder`, that's the shittiest Widget ever.
A separated class with a `ValueNotifier`-like thing or I don't know which state management solution, but never mix UI with how the data is fetched.

I know the app uses this pattern everywhere, but if I can stop the spreading of this 🦠, I prefer.